cdap-master-spi/src/main/java/io/cdap/cdap/master/spi/twill/SecureTwillPreparer.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.master.spi.twill;
+
+import org.apache.twill.api.TwillPreparer;
+import org.apache.twill.api.TwillRunnable;
+
+/**
+ * Extension interface for {@link TwillPreparer} to implement if it supports execution of a {@link TwillRunnable}
+ * under secure contexts.
+ */
+public interface SecureTwillPreparer extends TwillPreparer {
+  /**
+   * Runs the given runnable as a certain identity.
+   *
+   * @param runnableName name of the {@link TwillRunnable}
+   * @param identity the identity name to run as
+   * @return this {@link TwillPreparer}
+   */
+  SecureTwillPreparer withIdentity(String runnableName, String identity);
+
+  /**
+   * Runs the given runnable with an externally-provided secret.
```

Comment:
Generically, an external secret is a secret which is made available by the preparer to the runnable. Exactly how this secret is made available is up to preparer implementation.

Reply:
Actually it might be a bit cleaner to make this method take no arguments since the goal of this is to mount an sConf which is a singleton, so there might not be a need to mount multiple different secrets.